cinema/views.py

```diff
@@ -1,45 +1,107 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status, generics, mixins, viewsets
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
-from rest_framework import status
+from rest_framework.views import APIView
 
-from django.shortcuts import get_object_or_404
+from cinema.models import Genre, Actor, CinemaHall, Movie
+from cinema.serializers import MovieSerializer, GenreSerializer, ActorSerializer, CinemaHallSerializer
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
 
+class GenreList(APIView):
+    @staticmethod
+    def get(request):
+        genre = Genre.objects.all()
+        serializer = GenreSerializer(genre, many=True)
+        return Response(serializer.data)
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+    @staticmethod
+    def post(request):
+        serializer = GenreSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
 
-    if request.method == "POST":
-        serializer = MovieSerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+class GenreDetail(APIView):
+    @staticmethod
```

Comment:
Similarly, the `@staticmethod` decorator is not needed for the `get_object` method in `GenreDetail`. It should be an instance method to maintain consistency with Django's class-based views.